Draft: AggLayer documentation in Innovation & design section (#514)
0xPolygon/polygon-docs: docs/cdk/agglayer/agglayer-go.md

## Overview

The Agglayer Golang is a web service designed to receive zero-knowledge proofs (ZKP) from CDK chains, ensure they are valid, and send them to the L1 verifier contract for final verification.

!!! warning
    This service is being deprecated and replaced with a more robust and efficient [Rust implementation](agglayer-rs.md).

## Architecture

The AggLayer Golang architecture supports interoperability among connected CDK chains by using ZKPs to ensure safety. It uses a PostgreSQL database for storage and interacts with both L1 and L2 chains through configured RPC nodes.

The diagram below shows the full start-up, running, and shutdown sequence for the AggLayer Golang application and its components.

<center>
![CDK architecture](../../img/cdk/agglayer/agglayer-go.png)
</center>

## Get started

### Run locally with Docker

1. Clone the repository:

    ```bash
    git clone https://github.com/AggLayer/agglayer-go
    ```

2. Execute the following command:

    ```bash
    make run-docker
    ```

## Production set up

> 1. Ensure only one instance of Agglayer is running at a time.

Reviewer: ```suggestion
1. Ensure only one instance of the Agglayer is running at a time.
```

Author: updated over a previous update

## Overview

> The Agglayer Golang service is a web service designed to receive zero-knowledge proofs from various CDK chains, verify their soundness, and send them to L1 for final verification.

Reviewer: ```suggestion
The Agglayer Golang is a web service designed to receive zero-knowledge proofs from CDK chains, ensure they are valid, and send them to the L1 verifier contract for final verification.
```

!!! warning

>     This service is now deprecating in favor of the more robust and efficient [Rust implementation](agglayer-rs.md).

Reviewer: ```suggestion
    This service is being deprecated and replaced with a more robust and efficient [Rust implementation](agglayer-rs.md).
```

## Architecture

> The AggLayer Golang architecture supports interactions with multiple CDK chains for ZKP verification. 
> 
> It uses PostgreSQL database for storage and interacts with both L1 and L2 chains through configured RPC nodes.
> 
> The diagram below shows the full start up, running, and shutdown sequence for the application and its components.

Reviewer: ```suggestion
The AggLayer Golang architecture supports interoperability among connected CDK chains by using ZKPs to ensure safety. It uses PostgreSQL database for storage and interacts with both L1 and L2 chains through configured RPC nodes.

The diagram below shows the full start-up, running, and shutdown sequence for the AggLayer Golang application and its components.
```

<center>
![CDK architecture](../../img/cdk/agglayer/agglayer-go.png)
</center>

## Get started

### Run locally with Docker

1. Clone the repository:

    ```bash
    git clone https://github.com/AggLayer/agglayer-go
    ```

2. Execute the following command:

    ```bash
    make run-docker
    ```

## Production set up

1. Ensure only one instance of Agglayer is running at a time.
2. Use a containerized setup, or OS level service manager/monitoring system, for automatic restarts in the case of failures.

### Prerequisites

#### Hardware

> - For each CDK chain it's necessary to configure it's corresponding RPC node, synced with the target CDK. 

Reviewer: ```suggestion
- For each CDK chain it's necessary to configure its corresponding RPC node, synced with the target CDK. 
```
